konnectd/ui/src/containers/Login/Chooseaccount.js

```diff
@@ -0,0 +1,150 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import { FormattedMessage } from 'react-intl';
+
+import { withStyles } from '@material-ui/core/styles';
+import List from '@material-ui/core/List';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemText from '@material-ui/core/ListItemText';
+import ListItemAvatar from '@material-ui/core/ListItemAvatar';
+import Avatar from '@material-ui/core/Avatar';
+import Typography from '@material-ui/core/Typography';
+
+import { executeLogonIfFormValid, advanceLogonFlow } from '../../actions/login';
+import { ErrorMessage } from '../../errors';
+
+const styles = theme => ({
+  subHeader: {
+    marginBottom: theme.spacing(2)
+  },
+  message: {
+    marginTop: theme.spacing(2)
+  },
+  accountList: {
+    marginLeft: theme.spacing(-3),
+    marginRight: theme.spacing(-3)
+  },
+  accountListItem: {
+    paddingLeft: theme.spacing(3),
+    paddingRight: theme.spacing(3)
+  }
+});
+
+class Chooseaccount extends React.PureComponent {
+  componentDidMount() {
```

Comment:
I believe a [functional component + an effect hook](https://reactjs.org/docs/hooks-overview.html#effect-hook) would allow you to trash ugly js classes in favor of pure magnificent functions <3

Reply:
Thanks for the tip on effect hook 👼